sklearn/cluster/tests/test_k_medoids.py

```diff
@@ -0,0 +1,185 @@
+"""Testing for K-Medoids"""
+import numpy as np
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import raises
+
+from sklearn.metrics.pairwise import PAIRWISE_DISTANCE_FUNCTIONS
+from sklearn.metrics.pairwise import euclidean_distances
+from sklearn.datasets import load_iris
+
+from sklearn.cluster import KMedoids, KMeans
+
+
+@raises(ValueError)
+def test_kmedoids_fit_fails_n_clusters_is_zero():
+
+    np.random.seed(0)
+
+    X = np.random.rand(100, 5)
```

Comment:
You could reuse this toy `X` in several tests by initializing it outside the test functions.